baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -14,17 +16,48 @@
 from gevent.server import StreamServer
 from kombu.mixins import ConsumerMixin
 from kombu.transport.virtual import Channel
+from prometheus_client import Counter
+from prometheus_client import Gauge
+from prometheus_client import Histogram
 
 from baseplate import Baseplate
 from baseplate import RequestContext
 from baseplate.clients.kombu import KombuSerializer
+from baseplate.lib.prometheus_metrics import default_latency_buckets
 from baseplate.server.queue_consumer import HealthcheckCallback
 from baseplate.server.queue_consumer import make_simple_healthchecker
 from baseplate.server.queue_consumer import MessageHandler
 from baseplate.server.queue_consumer import PumpWorker
 from baseplate.server.queue_consumer import QueueConsumerFactory
 
 
+class AmqpConsumerPrometheusLabels(NamedTuple):
```

Comment:
Would a `dataclass` make sense here? Is this used as a type anywhere?

```python
import dataclasses
import typing

class AMPQConsumerPrometheusLabelsNT(typing.NamedTuple):
    amqp_address: str
    amqp_virtual_host: str
    amqp_exchange: str
    amqp_routing_key: str
 

@dataclasses.dataclass
class AMPQConsumerPrometheusLabels:
    amqp_address: str
    amqp_virtual_host: str
    amqp_exchange: str
    amqp_routing_key: str

labels_nt = AMPQConsumerPrometheusLabelsNT(amqp_address="amqp_address", amqp_virtual_host="amqp_virtual_host", amqp_exchange="amqp_exchange", amqp_routing_key="amqp_routing_key")
# print(labels_nt._fields)

labels_dc = AMPQConsumerPrometheusLabels(amqp_address="amqp_address", amqp_virtual_host="amqp_virtual_host", amqp_exchange="amqp_exchange", amqp_routing_key="amqp_routing_key")
# print(tuple(dataclasses.asdict(labels_dc).keys()))

assert labels_nt._fields == tuple(dataclasses.asdict(labels_dc).keys())
```

Reply:
Something to consider: https://stackoverflow.com/questions/51671699/data-classes-vs-typing-namedtuple-primary-use-cases